src/Storages/MergeTree/MergeTreePartition.cpp

```diff
@@ -56,7 +56,15 @@ namespace
         {
             UInt8 type = Field::Types::UInt128;
             hash.update(type);
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
```

Comment:
Thanks for moving the logic one level deeper. One natural question which arises is why only UInt128 values are affected but UInt256 (l. 69), UInt128 (l. 81) and Int256 (l. 87) are not. E.g. I find that 01891_partition_hash tests only with value `-1` for these other integer types which has equal LE and BE representations, meaning the test is unable to notice potential differences ... Maybe you could test on LE and BE machines with values that have different representations?

Reply:
Thanks Robert, I will take 3 weeks vacation from today, will work on it after I am back vacation. I converted the PR to Draft mode.

Reply:
I tried the following SQLs which use values other than -1, they both generate same partition IDs on both LE and BE machines. This means all types except UUID works the same way on both endian machines. So we only need to change UUID hashing.(So I rolled back the previous commit). This makes sense since UUID has the same byte order on both endian machines(which is expected), but we cannot directly use it to hash on BE machine since the hashing algorithm takes 256 bit integer value of UUID and the byte order must be reverted to make sure it uses the same integer value as the one on LE machine.
The SQLs are as following:
```
create table tab2 (i8 Int8, i16 Int16, i32 Int32, i64 Int64, i128 Int128, i256 Int256, u8 UInt8, 
u16 UInt16, u32 UInt32, u64 UInt64, u128 UInt128, u256 UInt256, s String, fs FixedString(33),
 a Array(UInt8), t Tuple(UInt16, UInt32), d Date, dt DateTime('Asia/Istanbul'), 
dt64 DateTime64(3, 'Asia/Istanbul'), dec128 Decimal128(3), dec256 Decimal256(4), 
lc LowCardinality(String)) engine = MergeTree
 PARTITION BY (i8, i16, i32, i64, i128, i256, u8, u16, u32, u64, u128, u256, s, fs, a, t, d, dt, dt64, dec128, dec256, lc) order by tuple();

insert into tab2 values (123, 12345, 1234567890, 1234567890000000000, 
123456789000000000000000000000000000000, 
123456789000000000000000000000000000000000000000000000000000000000000000000000, 123, 
12345, 1234567890, 1234567890000000000, 123456789000000000000000000000000000000, 
123456789000000000000000000000000000000000000000000000000000000000000000000000, 'a', 
'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', [1, 2, 3], (-1, -2), '2020-01-01', '2020-01-01 01:01:01', '2020-01-01 
01:01:01', '123.456', '78.9101', 'a');

select partition_id from system.parts where table = 'tab2' and database = currentDatabase();
```

Result:
```
┌─partition_id─────────────────────┐
│ 8129cf3d78627995c225a5dd3872a2b6 │
└──────────────────────────────────┘
```

Reply:
Sounds good, thanks. 

I added your example to 01891_partition_hash and merged from master to get some more recent CI results. Once builds/tests are green, I'll merge.